apache/airflow
# Hide logical date and run id in trigger UI form

This PR is an improvement based on the feedback raised in issue #34726 and optimizes the layout of the trigger UI form.

This PR changes:
- Run ID and Logic date are now in the folded section of generated JSON config - why? Because most users don't know what this is and (most probably) only advanced users know what to enter. Still for these advanced users it is needed and therefore in the collapsed section
- Both fields nevertheless are added to the legacy form as well if `AIRFLOW__WEBSERVER__SHOW_TRIGGER_FORM_IF_NO_PARAMS=True` is set
- The DAG documentation is now always expanded in the UI allowing to show instructions to the users.
- Update screenshot in the docs to new layout

How to test?
- Launch the UI and and trigger the DAG `example_params_ui_tutorial` and check the form
- Launch webserver with ENV `AIRFLOW__WEBSERVER__SHOW_TRIGGER_FORM_IF_NO_PARAMS=True` and trigger the DAG `example_python_operator`with config
- Check the Github old/new screenshow if you want to have a "sneak preview".

UI Before: 
![image](https://github.com/apache/airflow/assets/95105677/acb5f157-197d-4590-887b-46ab6849671c)

UI After:
![image](https://github.com/apache/airflow/assets/95105677/2681fe01-a363-4b48-9b86-19c08ed6fc58)

DAG Options when un-folded lower section:
![image](https://github.com/apache/airflow/assets/95105677/adcc8722-5f03-43c0-b3f6-3c6bbff508c2)


closes: #34726
